Ska/engarchive/update_server_sync.py

```diff
@@ -196,9 +196,9 @@ def get_row_from_archfiles(archfiles):
     # date like 2019-02-20T2109z, human-readable and Windows-friendly (no :) for a unique
     # identifier for this set of updates.
     date_id = get_date_id(DateTime(archfiles[0]['filetime']).fits)
-    row = {'filetime0': archfiles[0]['filetime'],
+    row = {'date_id': date_id,
```

Comment:
And sure.  Removing the explicit list in file_defs and replacing with this strategy to get the columns in the right order seems fine.  You have a better handle on astropy development to know which would be more stable in the long-term.  
And we presently don't see a driver for keeping the column order in file_defs for reference (which you might want if you thought this could vary based on environment).

Reply:
Our code should always be written to be tolerant of changes to column order or potentially new columns. My original implementation was the root problem, nothing more.

In astropy we strive for API stability, but sometimes allow cleaning up legacy design mistakes (or in these cases, designs that became mistakes due to language changes).